python/python/lance/dataset.py

```diff
@@ -587,6 +595,19 @@ def filter(self, filter: Union[str, pa.compute.Expression]) -> ScannerBuilder:
         self._filter = filter
         return self
 
+    def with_fragments(self, fragments: Optional[Iterable[LanceFragment]]) -> ScannerBuilder:
+        if fragments is not None:
+            inner_fragments = []
+            for f in fragments:
+                if isinstance(f, LanceFragment):
+                    inner_fragments.append(f._fragment)
+                else:
+                    raise TypeError("fragments must be an iterable of LanceFragment")
```

Comment:
Could you also add a message to tell user what the current type of `f` is , for easy debugging?